client/deposit_polygon.go

```diff
@@ -0,0 +1,373 @@
+	if err != nil {
+		return NetworkConfigResponse{}, err
+	}
+
+	return networkConfigResponse, nil
+}
+
+func isPresent(allowedTokens []Currency, currency Currency) bool {
+	for _, v := range allowedTokens {
+		if v == currency {
+			return true
+		}
+	}
+	return false
+}
+
+type CrossChainDepositRequest struct {
+	Amount                 string   `json:"amount"`
+	Currency               Currency `json:"currency"`
+	Network                Network  `json:"network"`
+	DepositBlockchainHash  string   `json:"deposit_blockchain_hash"`
+	DepositBlockchainNonce string   `json:"deposit_blockchain_nonce"`
+}
+
+
+func (c *Client) CrossChainDepositStart(ctx context.Context, crossChainDepositRequest CrossChainDepositRequest) (CryptoDepositResponse, error) {
+	err := c.CheckAuth()
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+
+	requestBody, err := json.Marshal(crossChainDepositRequest)
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, c.crossChainDepositStartURL.String(), bytes.NewReader(requestBody))
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+
+	req.Header.Set("Authorization", fmt.Sprintf("JWT %s", c.jwtToken))
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+	defer resp.Body.Close()
+
+	var cryptoDepositResponse CryptoDepositResponse
+	err = json.NewDecoder(resp.Body).Decode(&cryptoDepositResponse)
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+
+	return cryptoDepositResponse, nil
+}
+
+func (c *Client) SetAllowanceForPolygonNetwork(amount int) {
+	c.polygonNetworkAllowanceSet = false
+	c.polygonNetworkAllowance = amount
+}
+
+func (c *Client) DepositFromPolygonNetwork(
+	ctx context.Context,
+	ethAddress string,
+	ethPrivateKey string,
+	starkPublicKey string,
+	currency Currency,
+	amount float64,
+	
+) (CryptoDepositResponse, error) {
+	// check amount
+	if amount < 0 {
+		return CryptoDepositResponse{}, ErrInvalidAmount
+	}
+
+	// check if logged in or not
+	err := c.CheckAuth()
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+
+	// network config already setup
+
+	allowedTokens := c.polygonConfig.AllowedTokensForDeposit
+	contractAddress := c.polygonConfig.DepositContract
+
+	if !isPresent(allowedTokens, currency) {
+		return CryptoDepositResponse{}, ErrCoinNotFound
+	}
+
+	currentCoin := c.polygonConfig.Tokens[currency]
+	blockchainDecimal, err := strconv.Atoi(currentCoin.BlockchainDecimal)
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+
+	// contract already setup
+
+	balance, err := getTokenBalance(ctx, c.polygonClient, ethAddress, currency, currentCoin.BlockchainDecimal, currentCoin.TokenContract)
+	if err != nil {
+		return CryptoDepositResponse{}, err
+	}
+
+	if balance.Cmp(big.NewFloat(amount)) == -1 {
+		return CryptoDepositResponse{}, fmt.Errorf("current balance: %v", balance)
```

Comment:
make the error more verbose

Reply:
What more should I add in the error can you give an example?

Reply:
something like `available balance is less than deposited amount`